### PR TITLE
adding thumbnail capability for cover images

### PR DESCRIPTION
**Pull request by philipdouble**
Re-considered version of the auto-thumb generation.  

I would like to be able to remove the coverUpload mutation.  Would require being able to add a second param to singleUpload mutation for 'type' - this can be either 'library' or 'cover'.  

Right now all that this uploadCover does is call uploadToRepo with type set to 'cover':
async uploadCover(parent, { file }) {
    this.uploadToRepo(parent, { file, type: 'cover' });

    return true;
  }

Would love feedback on how to accomplish this - otherwise - good to go.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const typeDefs = gql`
   }
   type Mutation {
     singleUpload(file: Upload!): UploadedMediaResponse!
+    coverUpload(file: Upload!): UploadedMediaResponse!
```
Comment (shaheinm) [inline server.js]: I had considered simply making thumbnails out of every image that gets uploaded anyway... thoughts?

Comment (philipdouble) [reply server.js]: I don't think that you'll need it for all...  Like library items or what about profile images.

File: uploaders/GitHubUploader.js [hunk 2]
```diff
@@ -42,41 +43,52 @@ class GitHubUploader {
...
-  async deleteFileAfterCommit(file) {
-    return new Promise(function (resolve) {
-      console.info(`${file} committed and removed successfully`);
-      resolve(true);
-    });
+    return streamArray;
   }
 
-  async uploadToRepo(parent, { file }) {
+  async uploadToRepo(parent, { file, type = 'library' }) {
```
Comment (shaheinm) [inline uploaders/GitHubUploader.js]: You could console.log it or something if you want to keep it around in the code,  I guess.

**Review by shaheinm — APPROVED** [review]
Get rid of the ESLint problem and we're golden here... Thanks!

File: uploaders/GitHubUploader.js (hunk 2 above)
Comment (philipdouble) [reply uploaders/GitHubUploader.js]: removed

**Pull request merged**